rustls/src/crypto/mod.rs

```diff
@@ -417,6 +418,20 @@ pub trait SupportedKxGroup: Send + Sync + Debug {
     /// you can use [`NamedGroup::Unknown`].
     fn name(&self) -> NamedGroup;
 
+    /// FFDHE group the `SupportedKxGroup` operates in.
+    ///
+    /// Return `None` if this group is not a FFDHE one.
+    ///
+    /// The default implementation calls `FfdheGroup::from_named_group`: this function
+    /// is extremely linker-unfriendly so it is recommended all key exchange implementers
+    /// provide this function.
+    ///
+    /// `rustls::ffdhe_groups` contains suitable values to return from this,
+    /// for example [`rustls::ffdhe_groups::FFDHE2048`][crate::ffdhe_groups::FFDHE2048].
+    fn ffdhe_group(&self) -> Option<FfdheGroup<'static>> {
```

Comment:
Nit: I'd probably sort this before `name()`? (Also in the impls.)

Reply:
reordered